### PR TITLE
core: dump jobs list on sigrtmin+18 with 0x500

### PR DESCRIPTION
**Pull request by bluca**
Split out of https://github.com/systemd/systemd/pull/27890

### CODE REVIEW

File: src/shared/common-signal.h [hunk 1]
```diff
@@ -48,6 +48,8 @@ enum {
 
         COMMON_SIGNAL_COMMAND_MEMORY_PRESSURE = 0x300,
         COMMON_SIGNAL_COMMAND_MALLOC_INFO,
+
+        /* Private signals start at 0x500 */
```
Comment (poettering) [inline src/shared/common-signal.h:53]: maybe make that official:

```c
_COMMON_SIGNAL_COMMAND_PRIVATE_BASE = 0x500,
_COMMON_SIGNAL_COMMAND_PRIVATE_END = 0xFFF,
```

File: src/core/manager.h [hunk 2]
```diff
@@ -23,6 +23,9 @@ typedef struct Unit Unit;
 /* Enforce upper limit how many names we allow */
 #define MANAGER_MAX_NAMES 131072 /* 128K */
 
+/* On sigrtmin+18, private command */
+#define MANAGER_SIGNAL_COMMAND_DUMP_JOBS 0x500
```
Comment (poettering) [inline src/core/manager.h]: maybe define this as `_COMMON_SIGNAL_COMMAND_PRIVATE_BASE + 0` then

Comment (poettering) [reply src/core/manager.h]: also, maybe define as anonymous enum too

**Review by poettering — CHANGES_REQUESTED** [review]
lgtm

File: src/core/manager.h (hunk 2 above)
Comment (poettering) [reply src/core/manager.h]: or maybe define as:

```c
enum {
        MANAGER_SIGNAL_COMMAND_DUMP_JOBS = x0500
},

assert_cc(MANAGER_SIGNAL_COMMAND_DUMP_JOBS >= _COMMON_SIGNAL_COMMAND_PRIVATE_BASE);
assert_cc(MANAGER_SIGNAL_COMMAND_DUMP_JOBS <= _COMMON_SIGNAL_COMMAND_PRIVATE_END);
```

Comment (bluca) [reply src/core/manager.h]:
```
../src/core/manager.h:32:44: warning: comparison between ‘enum <anonymous>’ and ‘enum <anonymous>’ [-Wenum-compare]
   32 | assert_cc(MANAGER_SIGNAL_COMMAND_DUMP_JOBS >= _COMMON_SIGNAL_COMMAND_PRIVATE_BASE);
```

I'll go for the first suggestion

Comment (poettering) [reply src/core/manager.h]: oh, that's just dumb of gcc...

Comment (bluca) [reply src/core/manager.h]: actually can't even do that, all hell breaks loose as soon as I add common-signal.h to manager.h, I think there's some circular header shenanigans, the errors don't even make sense

```
In file included from ../src/core/manager.c:35:
../src/shared/common-signal.h:33:9: error: redeclaration of enumerator ‘_COMMON_SIGNAL_COMMAND_LOG_LEVEL_BASE’
   33 |         _COMMON_SIGNAL_COMMAND_LOG_LEVEL_BASE = 0x100,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/core/manager.h:132,
                 from ../src/core/emergency-action.h:36,
                 from ../src/core/unit.h:13,
                 from ../src/core/all-units.h:4,
                 from ../src/core/manager.c:24:
../src/shared/common-signal.h:33:9: note: previous definition of ‘_COMMON_SIGNAL_COMMAND_LOG_LEVEL_BASE’ with type ‘enum <anonymous>’
   33 |         _COMMON_SIGNAL_COMMAND_LOG_LEVEL_BASE = 0x100,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/shared/common-signal.h:34:9: error: redeclaration of enumerator ‘COMMON_SIGNAL_COMMAND_LOG_EMERG’
   34 |         COMMON_SIGNAL_COMMAND_LOG_EMERG       = _COMMON_SIGNAL_COMMAND_LOG_LEVEL_BASE + LOG_EMERG,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/shared/common-signal.h:34:9: note: previous definition of ‘COMMON_SIGNAL_COMMAND_LOG_EMERG’ with type ‘enum <anonymous>’
   34 |         COMMON_SIGNAL_COMMAND_LOG_EMERG       = _COMMON_SIGNAL_COMMAND_LOG_LEVEL_BASE + LOG_EMERG,

```

Comment (poettering) [reply src/core/manager.h]: common-signal.h is missing `#pragma once`, that is all. trivial fix...

Comment (bluca) [reply src/core/manager.h]: eheh, completely missed that - fixed

**Comment by poettering** [comment]
lgtm

File: src/core/manager.h [hunk 3]
```diff
@@ -23,6 +24,11 @@ typedef struct Unit Unit;
 /* Enforce upper limit how many names we allow */
 #define MANAGER_MAX_NAMES 131072 /* 128K */
 
+/* On sigrtmin+18, private commands */
+enum {
+        MANAGER_SIGNAL_COMMAND_DUMP_JOBS = _COMMON_SIGNAL_COMMAND_PRIVATE_BASE + 0,
```
Comment (poettering) [inline src/core/manager.h:29]: could even make this:

```c
enum {
        MANAGER_SIGNAL_COMMAND_DUMP_JOBS = _COMMON_SIGNAL_COMMAND_PRIVATE_BASE + 0,
        _MANAGER_SIGNAL_COMMAND_MAX,
};

assert_cc((int) _MANAGER_SIGNAL_COMMAND_MAX <= (int) _COMMON_SIGNAL_COMMAND_PRIVATE_END);
```

just to ensure that command nrs assigned here fit in the BASE/END range we define for it.

File: src/core/manager.h (hunk 2 above)
Comment (poettering) [reply src/core/manager.h]: > ../src/core/manager.h:32:44: warning: comparison between ‘enum <anonymous>’ and ‘enum <anonymous>’ [-Wenum-compare]
   32 | assert_cc(MANAGER_SIGNAL_COMMAND_DUMP_JOBS >= _COMMON_SIGNAL_COMMAND_PRIVATE_BASE);

btw, just cast the values to (int) before comparing, and the warning should go away

**Pull request merged**